api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1922,5 +1922,57 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
-	
+	/**
+	 * Get Concept by id or name convenience method
+	 * 
+	 * @param idOrMap
+	 * @return the found Concept
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public static Concept getConceptInfo(String idOrMap) throws APIException {
```

Comment:
@jnsereko This should also be resilient enough to also get concept by concept mapping. eg. 'CIEL:1234' 

So the naming becomes tricky here, I would keep it at:
`getConcept(String conceptRef)` or `getConcept(String id)` or `getConcept(String phrase)`

Since this method is being added to the service layer, I would not make it static hence:

```java

public Concept getConcept(String conceptRef)throws APIException {
         // some logic here
        return concept;
}
```